source/rules/no-ignored-replay-buffer.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * @license Use of this source code is governed by an MIT-style license that
+ * can be found in the LICENSE file at https://github.com/cartant/eslint-plugin-rxjs
+ */
+
+import { Rule } from "eslint";
+import * as es from "estree";
+import { getParent } from "../utils";
+
+const rule: Rule.RuleModule = {
+  meta: {
+    docs: {
+      category: "RxJS",
+      description:
+        "Disallows using `ReplaySubject`, `publishReplay` or `shareReplay` without specifying the buffer size.",
+      recommended: false
+    },
+    fixable: null,
+    messages: {
+      forbidden: "Ignoring the buffer size is forbidden."
+    },
+    schema: []
+  },
+  create: context => {
+    function report(
+      node: es.Node,
+      { arguments: args }: { arguments: es.Node[] }
+    ) {
+      if (!args || args.length === 0) {
+        context.report({
+          messageId: "forbidden",
+          node
+        });
+      }
+    }
+
+    return {
+      "NewExpression > Identifier[name='ReplaySubject']": (
+        node: es.Identifier
+      ) => {
+        const newExpression = getParent(node) as es.NewExpression;
+        report(node, newExpression);
+      },
+      "NewExpression MemberExpression Identifier[name='ReplaySubject']": (
+        node: es.Identifier
+      ) => {
+        const memberExpression = getParent(node) as es.MemberExpression;
+        const newExpression = getParent(memberExpression) as es.NewExpression;
+        report(node, newExpression);
+      },
+      "CallExpression Identifier[name=/(publishReplay|shareReplay)/]": (
```

Comment:
Might as well make this RegExp specific, too: `/^(publishReplay|shareReplay)$/`

Reply:
Because this selector does not use the direct-child combinator (`>`) I think it will match things it shouldn't. E.g. I think it will match this:

```ts
someFunction(publishReplay);
```

I think the selector needs to match only call expressions in which the expression is an identifier with a matching name.

Reply:
👍 I thought there was a reason for for not using the child combinator selector. 
I also added the child combinator to the 2nd selector (`NewExpression > MemberExpression > Identifier[name='ReplaySubject']`).